crates/miden-lib/src/account/interface/mod.rs

```diff
@@ -0,0 +1,384 @@
+    account_id: AccountId,
+    auth: Vec<AuthScheme>,
+    component_interfaces: BTreeSet<AccountComponentInterface>,
+}
+
+impl AccountInterface {
+    // CONSTRUCTORS
+    // --------------------------------------------------------------------------------------------
+
+    /// Creates a new [`AccountInterface`] instance from the provided account ID, authentication
+    /// schemes and account code.
+    pub fn new(account_id: AccountId, auth: Vec<AuthScheme>, code: &AccountCode) -> Self {
+        let component_interfaces = AccountComponentInterface::from_procedures(code.procedures());
+
+        Self { account_id, auth, component_interfaces }
+    }
+
+    // PUBLIC ACCESSORS
+    // --------------------------------------------------------------------------------------------
+
+    /// Returns a reference to the account ID.
+    pub fn id(&self) -> &AccountId {
+        &self.account_id
+    }
+
+    /// Returns the type of the reference account.
+    pub fn account_type(&self) -> AccountType {
+        self.account_id.account_type()
+    }
+
+    /// Returns true if the reference account can issue assets.
+    pub fn is_faucet(&self) -> bool {
+        self.account_id.is_faucet()
+    }
+
+    /// Returns true if the reference account is a regular.
+    pub fn is_regular_account(&self) -> bool {
+        self.account_id.is_regular_account()
+    }
+
+    /// Returns true if the reference account is public.
+    pub fn is_public(&self) -> bool {
+        self.account_id.is_public()
+    }
+
+    /// Returns a reference to the vector of used authentication schemes.
+    pub fn auth(&self) -> &Vec<AuthScheme> {
+        &self.auth
+    }
+
+    /// Returns a reference to the set of used component interfaces.
+    pub fn components(&self) -> &BTreeSet<AccountComponentInterface> {
+        &self.component_interfaces
+    }
+
+    /// checks if a note can be consumed against the current [AccountInterface] instance.
+    pub fn can_consume(&self, note: &Note) -> NoteAccountCompatibility {
+        let basic_wallet_notes = [p2id_commitment(), p2idr_commitment(), swap_commitment()];
+        let is_basic_wallet_note = basic_wallet_notes.contains(&note.script().hash());
+
+        if is_basic_wallet_note {
+            if self.component_interfaces.contains(&AccountComponentInterface::BasicWallet) {
+                return NoteAccountCompatibility::Yes;
+            }
+
+            // if the used interface vector doesn't contain neither the `BasicWallet` (which were
+            // checked above), nor the `Custom` account interfaces, then the only possible
+            // interfaces left in the vector are `BasicFungibleFaucet` and/or `RpoFalcon512`.
+            // Neither of them could consume the basic wallet note, so we could return `No` without
+            // checking the procedure hashes.
+            if !self.contains_non_standard_interface() {
+                return NoteAccountCompatibility::No;
+            }
+        }
```

Comment:
I would maybe change the logic here a bit:

1. First figure out if `note` is a well-known note (e.g., by using a `WellKnownNote` enum I mentioned in another comment).
2. If it is a well-known note, check if it can be consumed via something like `WellKnownNote::is_consumable_by()`.
3. If it isn't, run `verify_note_script_compatibility()` as we do now.